src/main/java/org/jenkinsci/plugins/github_branch_source/DefaultGitHubNotificationStrategy.java

```diff
@@ -0,0 +1,124 @@
+        Run<?, ?> build = notificationContext.getBuild();
+        Job<?, ?> job = notificationContext.getJob();
+        String url = null;
+        try {
+            if (null != build) {
+                url = DisplayURLProvider.get().getRunURL(build);
+            }
+            else if (null != job) {
+                url = DisplayURLProvider.get().getJobURL(job);
+            }
+        } catch (IllegalStateException e) {
+            listener.getLogger().println(
+                    "Can not determine Jenkins root URL. Commit status notifications are disabled "
+                            + "until a root URL is"
+                            + " configured in Jenkins global configuration.");
+        }
+        return url;
+    }
+
+    /**
+     * @since TODO
+     */
+    protected String generateMessage(GitHubNotificationContext notificationContext, TaskListener listener) {
+        Run<?, ?> build = notificationContext.getBuild();
+        if (null != build) {
+            Result result = build.getResult();
+            if (Result.SUCCESS.equals(result)) {
+                return Messages.GitHubBuildStatusNotification_CommitStatus_Good();
+            } else if (Result.UNSTABLE.equals(result)) {
+                return Messages.GitHubBuildStatusNotification_CommitStatus_Unstable();
+            } else if (Result.FAILURE.equals(result)) {
+                return Messages.GitHubBuildStatusNotification_CommitStatus_Failure();
+            } else if (Result.ABORTED.equals(result)) {
+                return Messages.GitHubBuildStatusNotification_CommitStatus_Aborted();
+            } else if (result != null) { // NOT_BUILT etc.
+                return Messages.GitHubBuildStatusNotification_CommitStatus_Other();
+            } else {
+                return Messages.GitHubBuildStatusNotification_CommitStatus_Pending();
+            }
+        }
+        return Messages.GitHubBuildStatusNotification_CommitStatus_Queued();
+    }
+
+    /**
+     * @since TODO
+     */
+    protected GHCommitState generateState(GitHubNotificationContext notificationContext, TaskListener listener) {
+        Run<?, ?> build = notificationContext.getBuild();
+        if (null != build) {
+            Result result = build.getResult();
+            if (Result.SUCCESS.equals(result)) {
+                return GHCommitState.SUCCESS;
+            } else if (Result.UNSTABLE.equals(result)) {
+                return GHCommitState.FAILURE;
+            } else if (Result.FAILURE.equals(result)) {
+                return GHCommitState.ERROR;
+            } else if (Result.ABORTED.equals(result)) {
+                return GHCommitState.ERROR;
+            } else if (result != null) { // NOT_BUILT etc.
+                return GHCommitState.ERROR;
+            }
+        }
+        return GHCommitState.PENDING;
+    }
+
+    /**
+     * @since TODO
+     */
+    protected boolean ignoreError(GitHubNotificationContext notificationContext, TaskListener listener) {
```

Comment:
would it make more sense to move this to a `getDefaultIgnoreError()` method on `GitHubNotificationContext`?